Extension/src/LanguageServer/extension.ts

```diff
@@ -42,8 +44,89 @@ let buildInfoCache: BuildInfo | null = null;
 const taskSourceStr: string = "C/C++";
 const cppInstallVsixStr: string = 'C/C++: Install vsix -- ';
 let taskProvider: vscode.Disposable;
+let codeActionProvider: vscode.Disposable;
 const intelliSenseDisabledError: string = "Do not activate the extension when IntelliSense is disabled.";
 
+type vcpkgDatabase = { [key: string]: string[] }; // Stored as <header file entry> -> [<port name>]
+let vcpkgDbPromise: Promise<vcpkgDatabase>;
+async function initVcpkgDatabase(): Promise<vcpkgDatabase> {
+    return new Promise((resolve, reject) => {
+        yauzl.open(util.getExtensionFilePath('VCPkgHeadersDatabase.zip'), { lazyEntries: true }, async (err? : Error, zipfile?: yauzl.ZipFile) => {
+            if (err) {
+                resolve({});
+                return;
+            }
+            zipfile.readEntry();
+            zipfile.on('entry', entry => {
+                if (entry.fileName !== 'VCPkgHeadersDatabase.txt') {
+                    resolve({});
```

Comment:
Actually, you shouldn't `resolve` the promise here. You should just ignore the entry.  But you probably want a handler for 'end' in case the entry was never encountered so you can resolve the promise as empty.

Reply:
Good catch!